extensions/shadcn-ui/src/search-components.tsx

```diff
@@ -0,0 +1,170 @@
+import { ActionPanel, Action, List, showToast, Toast } from "@raycast/api";
+import { useCachedPromise } from "@raycast/utils";
+import fetch, { type Response } from "node-fetch";
+import yaml from "js-yaml";
+
+const onRequestError = async () => {
+  await showToast({
+    style: Toast.Style.Failure,
+    title: "Request failed 🔴",
+    message: "Please try again later 🙏",
+  });
+};
+export default function Command() {
+  const { isLoading, data } = useCachedPromise(
+    async (url: string) => {
+      const response = await fetch(url);
+      return await parseFetchResponse(response);
+    },
+    ["https://ui.shadcn.com/api/component"],
+    {
+      keepPreviousData: true,
+      onError: onRequestError,
```

Comment:
Not sure about abstracting this error here and in `SearchListItem`. It means we have very generic error messages when in fact it would be nice where/why the error happens

Reply:
improved here d67ed4ffa8040a8a5b88540efb3e9546897e1ad1 please check 🙏🏼